src/test/java/onlinebookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,132 @@
+package onlinebookstore.controller;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.math.BigDecimal;
+import java.util.List;
+import onlinebookstore.dto.book.BookDto;
+import onlinebookstore.dto.book.CreateBookRequestDto;
+import onlinebookstore.exception.EntityNotFoundException;
+import onlinebookstore.repository.BookRepository;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@Sql(scripts = {"classpath:database/add-categories-to-database.sql",
+        "classpath:database/add-books-to-database.sql",
+        "classpath:database/add-books-categories-dependencies.sql"})
+@Sql(scripts = {"classpath:database/delete-all-from-database.sql"},
+        executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
```

Comment:
Better use SqlMergeMode(SqlMergeMode.MergeMode.MERGE)  because  you can have some problems if you have several SQL statement